testcases/PowerNVDump.py

```diff
@@ -135,7 +138,28 @@ def setup_test(self, dump_place="local"):
                 "ls -l /var/crash | grep '^d'| awk '{print $9}'")
         if dump_place == "net":
             self.crash_content = self.c.run_command(
-                "ssh root@%s \"ls -l %s | grep '^d'\" | awk '{print $9}'" % (self.dump_server_ip, self.dump_path))
+                "ssh -i %s root@%s \"ls -l %s | grep '^d'\" | awk '{print $9}'" % (self.rsa_path, self.dump_server_ip, self.dump_path))
+            log.debug("crash content is %s"%(self.crash_content))
+ 
+    def setup_pwdless_auth(self):
+        # Setup Pulickey authentication for Kdump over ssh and NFS
+        try:
+            self.c.run_command("ls %s"% self.rsa_path)
+            self.c.run_command("rm -f %s"% self.rsa_path)
+        except:
+            pass
+        try:
+            self.c.run_command("ssh-keygen -q -t rsa -f %s -P ''"% self.rsa_path)
+            try:
+                self.c.run_command("chmod 400 %s"% self.rsa_path)
+                self.c.run_command("sshpass -p %s ssh-copy-id -o \"StrictHostKeyChecking no\" -i %s root@%s" %(self.dump_server_pw, self.rsa_path, self.dump_server_ip))
+            except CommandFailed:
+                self.fail("Failed to copy public key to dump server")
+        except CommandFailed:
```

Comment:
why nested try,  have all 3 commands in single try and single exception with generic fail message